src/query/functions-v2/src/scalars/control.rs

```diff
@@ -173,4 +186,108 @@ pub fn register(registry: &mut FunctionRegistry) {
             ValueRef::Scalar(Some(_)) => Ok(Value::Scalar(true)),
         },
     );
+
+    fn cal_in<T: ArgType>(
+        lhs: ValueRef<T>,
+        rhs: ValueRef<ArrayType<T>>,
+    ) -> Result<Value<BooleanType>, String>
+    where
+        T::Scalar: HashtableKeyable,
+    {
+        // must be scalar in rhs, if rhs contains column, it'll be rewrited to `a = b1 or a = b2 or a = b3` in type_checker
```

Comment:
I think the assumption doesn't apply when the function is called explicitly by the user.

Reply:
Currently, all `col in (col1, col2, ... )` will be converted to `col = col1 or col = col2 ...` in the parse phase.

Reply:
Ok, let's throw errors as clickhouse did and let users use  `Expr::InList` or `contains` function instead (TBD).